File: 学习之路/dingTalkRobotCode/1.py
'''
                       _oo0oo_
                      o8888888o
                      88" . "88
                      (| -_- |)
                      0\  =  /0
                    ___/`---'\___
                  .' \\|     |// '.
                 / \\|||  :  |||// \
                / _||||| -:- |||||- \
               |   | \\\  - /// |   |
               | \_|  ''\---/''  |_/ |
               \  .-\__  '-'  ___/-. /
             ___'. .'  /--.--\  `. .'___
          ."" '<  `.___\_<|>_/___.' >' "".
         | | :  `- \`.;`\ _ /`;.`/ - ` : | |
         \  \ `_.   \_ __\ /__ _/   .-` /  /
     =====`-.____`.___ \_____/___.-`___.-'=====
                       `=---='


     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

           佛祖保佑       永不宕机     永无BUG

       佛曰:  
               写字楼里写字间，写字间里程序员；  
               程序人员写程序，又拿程序换酒钱。  
               酒醒只在网上坐，酒醉还来网下眠；  
               酒醉酒醒日复日，网上网下年复年。  
               但愿老死电脑间，不愿鞠躬老板前；  
               奔驰宝马贵者趣，公交自行程序员。  
               别人笑我忒疯癫，我笑自己命太贱；  
               不见满街漂亮妹，哪个归得程序员？

@Description: 
@Author: 石东昭
@Date: 2020-05-21 15:56:04
@LastEditors: shidongzhao
@LastEditTime: 2020-05-21 17:54:30
'''
# https://oapi.dingtalk.com/robot/send?access_token=b386b2008b69e98cbb3f14235b058f0adc29c17a55a6dec64dd6abef8bdf658e


import requests
import json
import time
import hmac
import hashlib
import base64
import urllib.parse

timestamp = str(round(time.time() * 1000))
print(time.time())
secret = 'SEC662eab928173979600e6b91e25433eacb28e6e3b63205f7ff471728685b463ac'
secret_enc = secret.encode('utf-8')
string_to_sign = '{}\n{}'.format(timestamp, secret)
string_to_sign_enc = string_to_sign.encode('utf-8')
hmac_code = hmac.new(secret_enc, string_to_sign_enc, digestmod=hashlib.sha256).digest()
sign = urllib.parse.quote_plus(base64.b64encode(hmac_code))

def dingmessage():
    webhook = 'https://oapi.dingtalk.com/robot/send?access_token=b386b2008b69e98cbb3f14235b058f0adc29c17a55a6dec64dd6abef8bdf658e&timestamp='+timestamp+'&sign='+sign
    header = {
        "Content-Type": 'application/json',
        "Charset": "UTF-8"
    }

    text = '上班下班要注意打卡'
    message = {
        'msgtype': 'text',
        "text": {
            "content": text
        },
        'at': {
            'isAtAll': True
        }
    }
    message_json = json.dumps(message)
    # info = requests.post(url=webhook, data=message_json, headers=header)

if __name__ == '__main__':
    dingmessage()

File: 学习之路/dingTalkRobotCode/1.js
/*
 *                   江城子 . 程序员之歌
 * 
 *               十年生死两茫茫，写程序，到天亮。
 *                   千行代码，Bug何处藏。
 *               纵使上线又怎样，朝令改，夕断肠。
 * 
 *               领导每天新想法，天天改，日日忙。
 *                   相顾无言，惟有泪千行。
 *               每晚灯火阑珊处，夜难寐，加班狂。
 * 
 * 
 * @Author: shdongzhao
 * @Date: 2020-05-21 16:39:20
 * @LastEditors: shidongzhao
 * @LastEditTime: 2020-05-21 18:12:14
 * @Description: 
 */
const Robot = require("dingtalk-robot-sdk")
const robot = new Robot({
    accessToken: 'b386b2008b69e98cbb3f14235b058f0adc29c17a55a6dec64dd6abef8bdf658e',
    secret: 'SEC662eab928173979600e6b91e25433eacb28e6e3b63205f7ff471728685b463ac',
});

const ActionCard = Robot.ActionCard;
const actionCard = new ActionCard();
actionCard.setTitle("乔布斯 20 年前想打造一间苹果咖啡厅，而它正是 Apple Store 的前身")
    .setText(`![screenshot](@lADOpwk3K80C0M0FoA)
### 乔布斯 20 年前想打造的苹果咖啡厅
Apple Store 的设计正从原来满满的科技感走向生活化，而其生活化的走向其实可以追溯到 20 年前苹果一个建立咖啡馆的计划`)
    .setHideAvatar(1).setBtnOrientation(1)
    .setSingleTitle("阅读全文")
    .setSingleURL("https://www.dingtalk.com/");
robot.send(actionCard)